projects/lights_control/src/lights_gpio.c

```diff
@@ -0,0 +1,212 @@
+#include "lights_gpio.h"
+#include "event_queue.h"
+#include "gpio.h"
+#include "gpio_mcu.h"
+#include "lights_events.h"
+#include "status.h"
+
+#define LIGHTS_HORN_PERIPHERAL 0
+#define NUM_LIGHTS_HORN_PERIPHERALS 1
+#define LIGHTS_STROBE_PERIPHERAL 0
+#define NUM_LIGHTS_STROBE_PERIPHERALS 1
+
+typedef enum {
+  LIGHTS_HIGH_BEAMS_RIGHT_PERIPHERAL = 0,
+  LIGHTS_HIGH_BEAMS_LEFT_PERIPHERAL,
+  NUM_LIGHTS_HIGH_BEAMS_PERIPHERALS
+} LightsHighBeamsPeripherals;
+
+typedef enum {
+  LIGHTS_LOW_BEAMS_RIGHT_PERIPHERAL = 0,
+  LIGHTS_LOW_BEAMS_LEFT_PERIPHERAL,
+  NUM_LIGHTS_LOW_BEAMS_PERIPHERALS
+} LightsLowBeamsPeripherals;
+
+typedef enum {
+  LIGHTS_DRL_RIGHT_PERIPHERAL = 0,
+  LIGHTS_DRL_LEFT_PERIPHERAL,
+  NUM_LIGHTS_DRL_PERIPHERALS
+} LightsDRLPeripherals;
+
+typedef enum {
+  LIGHTS_RIGHT_REAR_BRAKE_PERIPHERAL = 0,
+  LIGHTS_RIGHT_REAR_OUTER_BRAKE_PERIPHERAL,
+  LIGHTS_LEFT_REAR_BRAKE_PERIPHERAL,
+  LIGHTS_LEFT_REAR_OUTER_BRAKE_PERIPHERAL,
+  LIGHTS_CENTRE_BRAKE_PERIPHERAL,
+  NUM_LIGHTS_BRAKES_PERIPHERALS
+} LightsBrakesPeripherals;
+
+typedef enum {
+  LIGHTS_SIDE_LEFT_INDICATOR_PERIPHERAL = 0,
+  LIGHTS_LEFT_TURN_PERIPHERAL,
+  NUM_LIGHTS_FRONT_SIGNAL_LEFT_PERIPHERALS
+} LightsFrontSignalLeftPeripherals;
+
+typedef enum {
+  LIGHTS_SIDE_RIGHT_INDICATOR_PERIPHERAL = 0,
+  LIGHTS_RIGHT_TURN_PERIPHERAL,
+  NUM_LIGHTS_FRONT_SIGNAL_RIGHT_PERIPHERALS
+} LightsFrontSignalRightPeripherals;
+
+typedef enum {
+  LIGHTS_LEFT_REAR_OUTER_TURN_PERIPHERAL = 0,
+  LIGHTS_LEFT_REAR_TURN_PERIPHERAL,
+  NUM_LIGHTS_REAR_SIGNAL_LEFT_PERIPHERALS
+} LightsRearSignalLeftPeripherals;
+
+typedef enum {
+  LIGHTS_RIGHT_REAR_OUTER_TURN_PERIPHERAL = 0,
+  LIGHTS_RIGHT_REAR_TURN_PERIPHERAL,
+  NUM_LIGHTS_REAR_SIGNAL_RIGHT_PERIPHERALS
+} LightsRearSignalRightPeripherals;
+
+static const GPIOAddress s_horn_peripheral_addresses[NUM_LIGHTS_HORN_PERIPHERALS] = {
```

Comment:
What's the benefit of having so many separate variables? Why not just use a single config struct like in chaos?